src/GraphQL/Extensions/GraphQLBuilderExtensions.cs

```diff
@@ -583,6 +593,7 @@ public static IGraphQLBuilder AddGraphTypes(this IGraphQLBuilder builder, Assemb
         /// This is equivalent to calling <see cref="SchemaExtensions.RegisterTypeMappings(ISchema)"/>
         /// within the schema constructor.
         /// </remarks>
+        [RequiresUnreferencedCode("Please ensure that the graph types used by your schema and their constructors are not trimmed by the compiler.")]
         public static IGraphQLBuilder AddClrTypeMappings(this IGraphQLBuilder builder)
```

Comment:
This is important to have this here because the only time you'd use this method is when you are relying on this method to identify graph types in the assembly and link them to the clr types.  Well, with AOT/trimming, the entire graph type would be trimmed out (not just the constructor), so rather than getting an error message during schema initialization about a missing constructor, the clr type might just get mapped to an auto-registering type.

So this method can't be used unless the user manually references each graph type they intend to use beforehand or otherwise prevents trimming of those types.

I did not add a similar attribute to `AddGraphTypes` because most `Field<>` methods are marked with the attribute necessary to retain the constructor -- and via MSIL they contain the reference to the graph type.  It is only the query/mutation/subscription types that users are likely to have an issue with, and the worse-case scenario is that they get an error message that the constructor is missing; easy fix.  So this is a useful method without a dangerous side-effect.

Reply:
Note that this attribute will show a compile-time error in .NET 7 for `PublishAOT` assemblies (until suppressed) such as our example app.

Reply:
OK. I think that the experience of using annotations will come over time.